src/Fable.AST/Fable.fs

```diff
@@ -266,7 +266,7 @@ type KeyKind =
 type GetKind =
     | ByKey of KeyKind
     | TupleIndex of int
-    | UnionField of index: int * fieldType: Type
+    | UnionField of index: int * fieldType: Type * field: Field
```

Comment:
+1 for the major release 😉

Reply:
I tried to do it, but at that point I have the Union Type and the Field Index, but not the Case or Tag... It is probably somewhere in a pattern matching clause (since you cannot access field without matching on the case first) but the expression can be about anything...

Reply:
Damn, I thought something like the following would work but the problem is we don’t have the `tag` so we cannot select the union case. We do need to extend the AST for that. Well, major release it is :)

```fsharp
    let (|UnionFieldExt|_|) (com: Compiler) = function
        | Get(e,kind,_,_) ->
            match kind, e.Type with
            | UnionField(index, _), DeclaredType(entRef, _genArgs) -> 
                let ent = com.GetEntity(entRef)
                List.tryItem tag ent.UnionCases
                |> Option.bind (fun unionCase -> unionCase.UnionCaseFields |> List.tryItem index)
                |> Option.map (fun field -> Some(e, field))
            | _ -> None
        | _ -> None
```